framework/src/timesteppers/TimeStepper.C

```diff
@@ -184,10 +190,10 @@ TimeStepper::computeFailedDT()
   if (_dt <= _dt_min)
     mooseError("Solve failed and timestep already at or below dtmin, cannot continue!");
 
-  // cut the time step in a half
-  if (0.5 * _dt >= _dt_min)
-    return 0.5 * _dt;
-  else // (0.5 * _current_dt < _dt_min)
+  // cut the time step
+  if (_cutback_factor_at_failure * _dt >= _dt_min)
```

Comment:
This if else block could be replaced by this single line:
```
return std::max(_cutback_factor_at_failure * _dt, _dt_min);
```

Reply:
yeah it could be, but at some point, we probably should just accept working, valid code :)

Reply:
Ha, yeah, that should not hold up this PR. I've seen this antipattern before though, so I thought I point it out.